app/src/main/java/org/mozilla/fenix/compose/list/ListItem.kt

```diff
@@ -0,0 +1,338 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.compose.list
+
+import android.content.res.Configuration
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.RowScope
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.defaultMinSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.material.Icon
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.painter.Painter
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.Favicon
+import org.mozilla.fenix.compose.PrimaryText
+import org.mozilla.fenix.compose.SecondaryText
+import org.mozilla.fenix.theme.FirefoxTheme
+
+private val LIST_ITEM_HEIGHT = 56.dp
+
+private val ICON_SIZE = 24.dp
+
+/**
+ * List item used to display a label with an optional description text and
+ * an optional, interactable icon at the end.
+ *
+ * @param label The label in the list item.
+ * @param description An optional description text below the label.
+ * @param onClick Called when the user clicks on the item.
+ * @param endIconPainter [Painter] used to display a [ListItemIcon] at the end of the list item.
+ * @param endIconDescription Content description of the end icon.
+ * @param onEndIconClick Called when the user clicks on the end icon.
```

Comment:
I think we should just drop these `end` prefix. I think it's generally understood based on the top level kDoc that the icon goes to the end.

Reply:
Agreed. Updated